File: internal/domain/transaction.go
package domain

import (
	"time"
)

type Transaction struct {
	ID         uint
	Amount     float64
	Name       string
	Note       string
	CategoryID int64
	UserID     int64
	MadeAt     time.Time
	CreatedAt  time.Time
	UpdatedAt  time.Time

	RecurringTransaction *RecurringTransaction
}

File: src/db/models/transaction.py
from sqlalchemy import Column, Integer, String, Float, DateTime, BigInteger
from sqlalchemy.sql import func

from src.db.models import Base


class Transaction(Base):
    __tablename__ = "transactions"

    id = Column(Integer, primary_key=True, index=True)
    amount = Column(Float, nullable=False)
    name = Column(String, nullable=False)
    note = Column(String, default="")
    category_id = Column(BigInteger)
    user_id = Column(BigInteger, nullable=False)
    made_at = Column(DateTime(timezone=True), server_default=func.now())
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(
        DateTime(timezone=True), default=func.now(), onupdate=func.now()
    )
